### ISSUE TITLE
Add contribution model for application configuration

### ISSUE THREAD

**Issue by bdemers** [spring-io/initializr#1109]
I would love to be able to create a new project that contains an `application.yml` instead of an `application.properties` file.

Maybe as an option? `properties=yaml`

Or maybe `whiteSpaceBugs=true` :trollface: 
Joking aside, we commonly give instructions to rename app.props to app.yml after creating a new project and then provide example configuration in YAML format.

I can understand the desire to keep the number of permutations down, so feel free to close this issue!

Thanks again!

**Comment by wilkinsona** [comment]
It feels to me like something that the library could support but we'd choose not to use it on start.spring.io.

**Comment by snicoll** [comment]
Yeah although the current contributor right now create an empty file. If we support yaml-based config, It would make sense to offer a model where users can contribute keys and then the configuration is written in either format. If we decide not to use this on start.spring.io, I'd argue we have other areas of the library that needs improving with more impacts on our service.

**Comment by wilkinsona** [comment]
> If we decide not to use this on start.spring.io, I'd argue we have other areas of the library that needs improving with more impacts on our service.

I agree.

**Comment by snicoll** [comment]
I've repurposed this issue as an enhancement in the library.

**Comment by expe-elenigen** [comment]
I'm sorry if I highjack this ticket, but I feel the real question should be, why using `application.properties` as the default configuration file instead of `application.yml`?
The `properties` files came first, so I get it many people were used by this, but considering `yaml` is supported since a while and technically I just feel it's hard not to see that `properties` are way more verbose and they don't follow the [DRY principe](https://en.wikipedia.org/wiki/Don't_repeat_yourself). It's fine if some people insist on using `properties`, but it's still technically a better practice in my opinion adopt `yaml` files and this tool is, for now, sending the wrong message:
> Adopt properties files in your app!

I don't mind creating  a new ticket if it helps to consider this option, I was just feeling if my suggestion is took in consideration and adopted, then this ticket would lose a lot of interest.

**Comment by mhalbritter** [comment]
Superseded by #1682.
